lib/wallet/integration/src/Test/Integration/Framework/DSL.hs

```diff
@@ -535,6 +537,27 @@ expectErrorMessage want = either expectation wantedErrorButSuccess . snd
         RawClientError val    -> BL8.unpack val
         HttpException err     -> show err
 
+-- | Expect an error identified by its code.
+expectErrorCode
+    :: (HasCallStack, MonadIO m, Show a)
+    => String
```

Comment:
Could we not
```suggestion
    => ApiErrorInfo
```
and compare the result of `decodeErrorInfo` to the expected value?

Reply:
I have intentionally avoided it, as our API contract is string value, not the ApiInfo type

Reply:
👌 I could see arguments for 1) not hardcoding the strings 2) also being able to write expectations for the "payload" of `ApiErrorInfo` when such exists, but "API contract is string value" — that's a neat focus so I'm not against it
cc @jonathanknowles

Reply:
I can imagine the situation where `ApiErrorInfo` <-> `String` serialization logic is wrong, but an integration test which uses `ApiErrorInfo` won't capture this as a failure.  Ultimately, every value except that string `transaction_is_too_big` mentioned in the swagger.yaml is a violation of a contract, so I prefer to stay as close to it as possible.